File: src/engine/commands/admin/additemcommand.cpp
#include "additemcommand.h"

#include "item.h"


#define super AdminCommand

AddItemCommand::AddItemCommand(QObject *parent) :
    super(parent) {

    setDescription("Add an item to the current room.\n"
                   "\n"
                   "Usage: add-item <item-name>");
}

AddItemCommand::~AddItemCommand() {
}

void AddItemCommand::execute(Character *player, const QString &command) {

    super::prepareExecute(player, command);

    if (!assertWordsLeft("Usage: add-item <item-name>")) {
        return;
    }

    QString itemName = takeWord();

    Item *item = new Item(realm());
    item->setName(itemName);
    currentRoom()->addItem(item);

    send("Item %1 added.", itemName);
}

File: data/commands/admin/additemcommand.js
function AddItemCommand() {
    AdminCommand.call(this);

    this.setDescription("Add an item to the current room.\n" +
                        "\n" +
                        "Usage: add-item <item-name>");
}

AddItemCommand.prototype = new AdminCommand();
AddItemCommand.prototype.constructor = AddItemCommand;

AddItemCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    if (!this.assertWordsLeft("Usage: add-item <item-name>")) {
        return;
    }

    var itemName = this.takeWord();

    var item = Realm.createObject("Item");
    item.name = itemName;
    this.currentRoom.addItem(item);

    this.send("Item %1 added.", itemName);
};

CommandRegistry.registerAdminCommand("add-item", new AddItemCommand());
